src/commands/utils/featureRegistrations.ts

```diff
@@ -0,0 +1,68 @@
+import { FeatureClient } from "@azure/arm-features";
+import { longRunning } from "./host";
+
+const MAX_RETRIES = 5;
+const RETRY_DELAY_MS = 5000; // 5 seconds
+
+export type MultipleFeatureRegistration = {
+    resourceProviderNamespace: string;
+    featureName: string;
+};
+
+export type FeatureRegistrationResult = {
+    resourceProviderNamespace: string;
+    featureName: string;
+    registrationStatus: string;
+};
+
+export async function createFeatureRegistrationsWithRetry(
+    featureClient: FeatureClient,
+    resourceProviderNamespace: string,
+    featureName: string,
+): Promise<FeatureRegistrationResult> {
+    let retries = 0;
+    let registrationStatus = "Registering";
+    //register the feature
+    featureClient.features.register(resourceProviderNamespace, featureName);
+    do {
+        // get the registration status
+        const featureRegistrationResult = await featureClient.features.get(resourceProviderNamespace, featureName);
+        const result = featureRegistrationResult.properties?.state || "Failed";
+        if (result === "Registered") {
+            registrationStatus = "Registered";
+            break;
+        } else if (result === "Registering" && retries < MAX_RETRIES) {
+            // if the feature is still registering and we haven't reached the max retries, wait and try again
+            registrationStatus = "Registering";
+            retries++;
+            await new Promise((resolve) => setTimeout(resolve, RETRY_DELAY_MS));
+        } else {
+            // if the feature registration failed or we reached the max retries but still feature registration status is registering, throw an error
+            throw new Error(
+                `Failed to register the preview feature ${featureName} for ${resourceProviderNamespace}. Current state: ${result}`,
+            );
+        }
+    } while (registrationStatus === "Registering" && retries < MAX_RETRIES);
```

Comment:
💡  might be in this or other PR but we should `enum` state this because these whole string comparison will lead to scattered logic handling. If its easy we should do this here please.